src/Core.flix

```diff
@@ -255,6 +256,52 @@ namespace Flixball {
                     List.find(match (foundId, _) -> foundId == id) |>
                     Option.map(match (_, pos) -> pos)
 
+            ///
+            /// Each line of the string is a row. If lines are of different
+            /// lengths they will be interpreted as being right-padded with
+            /// spaces. The board will have an surrounding edge added. Anything
+            /// not listed below will be interpreted as a wall. Players are
+            /// numbered from 0, top-to-buttom, left-to-right. 
+            ///
+            /// " " = empty tile
+            /// "#" = wall tile
+            /// {"<","^",">","v"} = player tile
+            ///
+            pub def fromString(s: String): Board = region r {
+                let lines = String.lines(s);
+
+                let rows = List.length(lines);
+                let cols = lines |>
+                    List.map(String.length) |>
+                    List.maximum |>
+                    Option.getWithDefault(0);
+                let tileMap = new MutMap(r);
+
+                let availableId = ref 0 @ r;
+                def freshId() = {
+                    let id = deref availableId;
+                    availableId := id + 1;
+                    id
+                };
+
+                foreach((line, row) <- lines |> List.enumerator(r)) {
+                    let cells = String.toList(line);
+                    foreach((cell, col) <- cells |> List.enumerator(r)) {
+                        def addTile(tile) = MutMap.put!((row, col), tile, tileMap);
+                        match cell {
+                            case ' ' => () // do nothing
+                            case '<' => addTile(Person(freshId(), West))
+                            case '^' => addTile(Person(freshId(), North))
+                            case '>' => addTile(Person(freshId(), East))
+                            case 'v' => addTile(Person(freshId(), South))
+                            case _   => addTile(Wall)
```

Comment:
I think it may be better to report an error in this case, since it may represent a typo on the part of the user.

Reply:
fixed